devito/interpolators.py

```diff
@@ -0,0 +1,106 @@
+        super(LinearInterpolator, self).__init__(point_symbols, dimensions, 1)
+
+    @property
+    def coefficients(self):
+        """Symbolic expression for the coefficients for sparse point
+        interpolation according to:
+        https://en.wikipedia.org/wiki/Bilinear_interpolation.
+
+        :returns: List of coefficients, eg. [b_11, b_12, b_21, b_22]
+        """
+        # Grid indices corresponding to the corners of the cell
+        x1, y1, z1, x2, y2, z2 = sympy.symbols('x1, y1, z1, x2, y2, z2')
+        # Coordinate values of the sparse point
+        px, py, pz = self.point_symbols
+        if self.ndim == 2:
+            A = sympy.Matrix([[1, x1, y1, x1*y1],
+                              [1, x1, y2, x1*y2],
+                              [1, x2, y1, x2*y1],
+                              [1, x2, y2, x2*y2]])
+
+            p = sympy.Matrix([[1],
+                              [px],
+                              [py],
+                              [px*py]])
+
+            # Map to reference cell
+            x, y = self.dimensions
+            reference_cell = {x1: 0, y1: 0, x2: x.spacing, y2: y.spacing}
+
+        elif self.ndim == 3:
+            A = sympy.Matrix([[1, x1, y1, z1, x1*y1, x1*z1, y1*z1, x1*y1*z1],
+                              [1, x1, y1, z2, x1*y1, x1*z2, y1*z2, x1*y1*z2],
+                              [1, x1, y2, z1, x1*y2, x1*z1, y2*z1, x1*y2*z1],
+                              [1, x1, y2, z2, x1*y2, x1*z2, y2*z2, x1*y2*z2],
+                              [1, x2, y1, z1, x2*y1, x2*z1, y1*z1, x2*y1*z1],
+                              [1, x2, y1, z2, x2*y1, x2*z2, y1*z2, x2*y1*z2],
+                              [1, x2, y2, z1, x2*y2, x2*z1, y2*z1, x2*y2*z1],
+                              [1, x2, y2, z2, x2*y2, x2*z2, y2*z2, x2*y2*z2]])
+
+            p = sympy.Matrix([[1],
+                              [px],
+                              [py],
+                              [pz],
+                              [px*py],
+                              [px*pz],
+                              [py*pz],
+                              [px*py*pz]])
+
+            # Map to reference cell
+            x, y, z = self.dimensions
+            reference_cell = {x1: 0, y1: 0, z1: 0, x2: x.spacing,
+                              y2: y.spacing, z2: z.spacing}
+        else:
+            raise NotImplementedError('Interpolation coefficients not implemented '
+                                      'for %d dimensions.' % self.grid.dim)
+
+        A = A.subs(reference_cell)
+        coeffs = A.inv().T.dot(p)
+        return coeffs
+
+
+class LanczosInterpolator(GenericInterpolator):
+    def __init__(self, point_symbols, dimensions, a):
+        super(LanczosInterpolator, self).__init__(point_symbols, dimensions, a)
+
+    @property
+    def coefficients(self):
+        x, a = sympy.symbols("x a")
+        f = sympy.sinc(x)*sympy.sinc(x/a)
+        vectors = []
+        for dim in self.point_symbols:
```

Comment:
It looks like self.point_symbols is always (px,py,pz) even for a 2D grid.  This causes this code to produce incorrect output when generating coefficients for 2D grids.

The real problem is probably that len(self.point_symbols) != ndim, though.